segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLogTests.java

```diff
@@ -82,41 +80,40 @@
 public abstract class BookKeeperLogTests extends DurableDataLogTestBase {
     //region Setup, Config and Cleanup
 
-    private static final AtomicBoolean SECURE_BK = new AtomicBoolean();
-
     private static final int CONTAINER_ID = 9999;
     private static final int WRITE_COUNT = 500;
     private static final int BOOKIE_COUNT = 1;
     private static final int THREAD_POOL_SIZE = 3;
     private static final int MAX_WRITE_ATTEMPTS = 5;
     private static final int MAX_LEDGER_SIZE = WRITE_MAX_LENGTH * Math.max(10, WRITE_COUNT / 20);
 
-    private static final AtomicReference<BookKeeperServiceRunner> BK_SERVICE = new AtomicReference<>();
-    private static final AtomicInteger BK_PORT = new AtomicInteger();
-
     @Rule
-    public Timeout globalTimeout = Timeout.seconds(100 * TIMEOUT.getSeconds());
+    public Timeout globalTimeout = Timeout.seconds(100);
     private final AtomicReference<BookKeeperConfig> config = new AtomicReference<>();
     private final AtomicReference<CuratorFramework> zkClient = new AtomicReference<>();
     private final AtomicReference<BookKeeperLogFactory> factory = new AtomicReference<>();
+    private final AtomicBoolean secureBk = new AtomicBoolean();
+    private final AtomicReference<BookKeeperServiceRunner> bkService = new AtomicReference<>();
 
     /**
      * Start BookKeeper once for the duration of this class. This is pretty strenuous, so in the interest of running time
      * we only do it once.
      */
-    public static void setUpBookKeeper(boolean secure) throws Exception {
+    public void setUpBookKeeper(boolean secure) throws Exception {
         // Pick a random port to reduce chances of collisions during concurrent test executions.
-        SECURE_BK.set(secure);
-        BK_PORT.set(TestUtils.getAvailableListenPort());
+        secureBk.set(secure);
+        String testId = Long.toHexString(System.nanoTime());
+        int zkPort = TestUtils.getAvailableListenPort();
         val bookiePorts = new ArrayList<Integer>();
         for (int i = 0; i < BOOKIE_COUNT; i++) {
             bookiePorts.add(TestUtils.getAvailableListenPort());
         }
 
+        String ledgersPath = "/pravega/bookkeeper/ledgers/" + testId;
         val runner = BookKeeperServiceRunner.builder()
                                             .startZk(true)
-                                            .zkPort(BK_PORT.get())
-                                            .ledgersPath("/pravega/bookkeeper/ledgers")
+                                            .zkPort(zkPort)
+                                            .ledgersPath(ledgersPath)
```

Comment:
Haven't you used ledgers path + zkPort before? Should we use the same here? This is what I'm thinking about:

https://github.com/pravega/pravega/pull/4999/files#diff-7cb38ef4f69c054db7f40fc17995e399cfbc2c96cc35ec1e5d40a969e3bb3aedR62

Reply:
From the above:
```
 String ledgersPath = "/pravega/bookkeeper/ledgers/" + testId;
 ```
 Where testId is:
 ```
 String testId = Long.toHexString(System.nanoTime());
 ```
 This serves much the same function. Either way it ensures that parallel tests won't collide in their directories.